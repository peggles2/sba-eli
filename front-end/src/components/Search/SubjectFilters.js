import React, { Component } from 'react';
import { Form } from 'semantic-ui-react'

export default class SubjectFilters extends Component {
  render () {
    const subjectFilters = [
      {key: 'finance', text: 'Finance'},
      {key: 'communications', text: 'Communications'},
      {key: 'management', text: 'Management'},
      {key: 'marketing', text: 'Marketing'},
      {key: 'sales', text: 'Sales'},
      {key: 'strategy', text: 'Strategy'},
    ]

    return subjectFilters.map(filter => {
      return (
        <Form.Checkbox name='subject'
                      value={filter.key}
                      key={filter.key}
                      id={'subject_filter_' +  filter.key}
                      label={{children: filter.text}}/>
      )
    })
  }
}
