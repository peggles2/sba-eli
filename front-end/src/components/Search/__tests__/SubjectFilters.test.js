import React from 'react';
import { shallow } from 'enzyme';
import { Form } from 'semantic-ui-react';
import SubjectFilters from '../SubjectFilters';

describe('SubjectFilters', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<SubjectFilters/>);

    expect(wrapper).toMatchSnapshot();
  });
  
  it('should render 6 <Checkboxes> for 6 Subjects', () => {
    const wrapper = shallow(<SubjectFilters/>);

    expect(wrapper.find(Form.Checkbox).length).toEqual(6);
  });

  it('should render the checkbox with the expected data from the filter', () => {
    const wrapper = shallow(<SubjectFilters/>);

    expect(wrapper.find(Form.Checkbox).get(0).key).toEqual('finance');
    expect(wrapper.find(Form.Checkbox).get(0).props.value).toEqual('finance');
    expect(wrapper.find(Form.Checkbox).get(0).props.id).toEqual('subject_filter_finance');
    expect(wrapper.find(Form.Checkbox).get(0).props.label.children).toEqual('Finance');
  });
});
