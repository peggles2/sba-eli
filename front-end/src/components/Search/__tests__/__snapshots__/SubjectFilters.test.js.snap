// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubjectFilters should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SubjectFilters />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "finance",
    "nodeType": "function",
    "props": Object {
      "as": [Function],
      "control": [Function],
      "id": "finance",
      "label": Object {
        "children": "Finance",
      },
      "name": "subject",
      "value": "finance",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "finance",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "finance",
        "label": Object {
          "children": "Finance",
        },
        "name": "subject",
        "value": "finance",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "communications",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "communications",
        "label": Object {
          "children": "Communications",
        },
        "name": "subject",
        "value": "communications",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "management",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "management",
        "label": Object {
          "children": "Management",
        },
        "name": "subject",
        "value": "management",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "marketing",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "marketing",
        "label": Object {
          "children": "Marketing",
        },
        "name": "subject",
        "value": "marketing",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "sales",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "sales",
        "label": Object {
          "children": "Sales",
        },
        "name": "subject",
        "value": "sales",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    Object {
      "instance": null,
      "key": "strategy",
      "nodeType": "function",
      "props": Object {
        "as": [Function],
        "control": [Function],
        "id": "strategy",
        "label": Object {
          "children": "Strategy",
        },
        "name": "subject",
        "value": "strategy",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
